managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -179,7 +179,16 @@ public void addComplete(int rc, final LedgerHandle lh, long entryId, Object ctx)
     public void safeRun() {
         // Remove this entry from the head of the pending queue
         OpAddEntry firstInQueue = ml.pendingAddEntries.poll();
-        checkArgument(this == firstInQueue);
+        if (firstInQueue == null) {
+            // The pending op might been polled by others such as cleanup pending op when create Ledger failed.
+            this.recycle();
+            return;
+        }
+        if (this != firstInQueue) {
+            firstInQueue.failed(new ManagedLedgerException("Unexpected add entry op when complete the add entry op."));
+            this.recycle();
+            return;
+        }
```

Comment:
How about `ReferenceCountUtil.release(data);` in these cases? would that be needed to prevent memory leaks? `this.recycle` doesn't handle releasing of `data`.

Reply:
Good point! Fixed.